client/src/main/java/io/pravega/client/netty/impl/ConnectionPoolImpl.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    @Override
+    public int getActiveChannelCount() {
+        return (int) this.channelGroup.stream()
+                                      .filter(Channel::isActive)
+                                      .peek(ch -> log.debug("Channel with id {} localAddress {} and remoteAddress {} is active.", ch.id(),
+                                                            ch.localAddress(), ch.remoteAddress()))
+                                      .count();
+    }
+
+    @VisibleForTesting
+    public List<Channel> getActiveChannels() {
+        return this.channelGroup.stream().filter(Channel::isActive).collect(Collectors.toList());
+    }
+
+    /**
+     * Establish a new connection to the Pravega Node.
+     * @param location The Pravega Node Uri
+     * @return A future, which completes once the connection has been established, returning a SessionHandler that can be used to create
+     * sessions on the connection.
+     */
+    private CompletableFuture<SessionHandler> establishConnection(PravegaNodeUri location) {
+        final AppendBatchSizeTracker batchSizeTracker = new AppendBatchSizeTrackerImpl();
+        final SessionHandler handler = new SessionHandler(location.getEndpoint(), batchSizeTracker);
+        final Bootstrap b = getNettyBootStrap().handler(getChannelInitializer(location, batchSizeTracker, handler));
+
+        // Initiate Connection.
+        final CompletableFuture<SessionHandler> connectionComplete = new CompletableFuture<>();
+        try {
+            b.connect(location.getEndpoint(), location.getPort()).addListener((ChannelFutureListener) future -> {
+                if (future.isSuccess()) {
+                    //since ChannelFuture is complete future.channel() is not a blocking call.
+                    Channel ch = future.channel();
+                    log.debug("Connect operation completed for channel:{}, local address:{}, remote address:{}",
+                              ch.id(), ch.localAddress(), ch.remoteAddress());
+                    channelGroup.add(ch); // Once a channel is closed the channel group implementation removes it.
+                    connectionComplete.complete(handler);
+                } else {
+                    connectionComplete.completeExceptionally(new ConnectionFailedException(future.cause()));
+                }
+            });
+        } catch (Exception e) {
+            connectionComplete.completeExceptionally(new ConnectionFailedException(e));
+        }
+
+        final CompletableFuture<Void> channelRegisteredFuture = new CompletableFuture<>(); //to track channel registration.
+        handler.completeWhenRegistered(channelRegisteredFuture);
+
+        return connectionComplete.thenCombine(channelRegisteredFuture, (sessionHandler, v) -> sessionHandler);
+    }
+
+    /**
+     * Create {@link Bootstrap}.
+     */
+    private Bootstrap getNettyBootStrap() {
```

Comment:
Please be consistent with casing. Either `Bootstrap` or `BootStrap`

Reply:
fixed it.